internal/plugin/connectors/tcp/mssql/connection_details_test.go

```diff
@@ -114,6 +117,10 @@ func TestNewConnectionDetails(t *testing.T) {
 	}
 }
 
+func TestDefaultSSLModeExists(t *testing.T) {
+	assert.NotNil(t, sslModeToBaseParams[string(defaultSSLMode)])
```

Comment:
`assert.NotNil` only works for value that can be nil. This test won't fail for an empty string. In fact, I just spotted that there's a few places in the codebase where `assert.NotNil` is being used for string values :'(. I'll create a card to fix those tests

Reply:
I think we need to use `assert.NotEmpty`

Reply:
Good catch - I'll create an issue for that and correct it here

Reply:
Corrected here, and new issue created - #1196